custom_containers/pytorch/mnist/trainer/mnist.py

```diff
@@ -0,0 +1,195 @@
+
+
+def test(args, model, device, test_loader, epoch):
+  model.eval()
+  test_loss = 0
+  correct = 0
+  with torch.no_grad():
+    for data, target in test_loader:
+      data, target = data.to(device), target.to(device)
+      output = model(data)
+      test_loss += F.nll_loss(
+          output, target, size_average=False).item()  # sum up batch loss
+      pred = output.max(
+          1, keepdim=True)[1]  # get the index of the max log-probability
+      correct += pred.eq(target.view_as(pred)).sum().item()
+
+  test_loss /= len(test_loader.dataset)
+  print('\nTest set: Average loss: {:.4f}, Accuracy: {}/{} ({:.0f}%)\n'.format(
+      test_loss, correct, len(test_loader.dataset),
+      100. * correct / len(test_loader.dataset)))
+
+  # Uses hypertune to report metrics for hyperparameter tuning.
+  hpt = hypertune.HyperTune()
+  hpt.report_hyperparameter_tuning_metric(
+      hyperparameter_metric_tag='my_loss',
+      metric_value=test_loss,
+      global_step=epoch)
+
+
+def main():
+  # Training settings
+  parser = argparse.ArgumentParser(description='PyTorch MNIST Example')
+  parser.add_argument(
+      '--batch-size',
+      type=int,
+      default=64,
+      metavar='N',
+      help='input batch size for training (default: 64)')
+  parser.add_argument(
+      '--test-batch-size',
+      type=int,
+      default=1000,
+      metavar='N',
+      help='input batch size for testing (default: 1000)')
+  parser.add_argument(
+      '--epochs',
+      type=int,
+      default=10,
+      metavar='N',
+      help='number of epochs to train (default: 10)')
+  parser.add_argument(
+      '--lr',
+      type=float,
+      default=0.01,
+      metavar='LR',
+      help='learning rate (default: 0.01)')
+  parser.add_argument(
+      '--momentum',
+      type=float,
+      default=0.5,
+      metavar='M',
+      help='SGD momentum (default: 0.5)')
+  parser.add_argument(
+      '--no-cuda',
+      action='store_true',
+      default=False,
+      help='disables CUDA training')
+  parser.add_argument(
+      '--seed',
+      type=int,
+      default=1,
+      metavar='S',
+      help='random seed (default: 1)')
+  parser.add_argument(
+      '--log-interval',
+      type=int,
+      default=10,
+      metavar='N',
+      help='how many batches to wait before logging training status')
+  parser.add_argument(
+      '--model-dir',
+      default=None,
+      help='The directory to store the model')
+
+  args = parser.parse_args()
+  use_cuda = not args.no_cuda and torch.cuda.is_available()
+
+  torch.manual_seed(args.seed)
+
+  device = torch.device('cuda' if use_cuda else 'cpu')
+
+  kwargs = {'num_workers': 1, 'pin_memory': True} if use_cuda else {}
+  train_loader = torch.utils.data.DataLoader(
+      datasets.MNIST(
+          '../data',
+          train=True,
+          download=True,
+          transform=transforms.Compose([
+              transforms.ToTensor(),
+              transforms.Normalize((0.1307,), (0.3081,))
```

Comment:
nit: what are 0.1307 and 0.3081 ? Can we add a comment to clarify. ?

Reply:
Done